core/src/display_object/movie_clip.rs

```diff
@@ -4063,6 +4209,59 @@ impl<'gc, 'a> MovieClipData<'gc> {
         Ok(())
     }
 
+    #[inline]
+    fn get_registered_character_by_id(
+        &mut self,
+        context: &mut UpdateContext<'_, 'gc>,
+        id: CharacterId,
+    ) -> Result<Character<'gc>, Error> {
+        let library_for_movie = context.library.library_for_movie(self.movie());
+
+        if let Some(library) = library_for_movie {
+            if let Some(character) = library.character_by_id(id) {
+                return Ok(character.clone());
+            }
+        }
+        Err(Error::InvalidSwf(swf::error::Error::invalid_data(
+            "message",
+        )))
+    }
+
+    fn register_export(
+        &mut self,
+        context: &mut UpdateContext<'_, 'gc>,
+        id: CharacterId,
+        name: &AvmString<'gc>,
+        movie: Arc<SwfMovie>,
+    ) {
+        let library = context.library.library_for_movie_mut(movie);
+        library.register_export(id, *name);
+
+        // TODO: do other types of Character need to know their exported name?
+        if let Some(character) = library.character_by_id(id) {
+            if let Character::MovieClip(movie_clip) = character {
+                *movie_clip
+                    .0
+                    .read()
+                    .static_data
+                    .exported_name
+                    .write(context.gc_context) = Some(*name);
+            } else {
+                tracing::debug!(
```

Comment:
This `debug!` feels like it should be a `warn!` or at least `info!` unless it's common enough that that would be cause spam

Reply:
I asked him to change it to debug or info before 😅 whoops